lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -145,6 +174,27 @@ defmodule OliWeb.Components.Delivery.Actions do
     end
   end
 
+  def handle_event("re_enroll", _params, socket) do
+    %{section: section, user: user} = socket.assigns
+
+    case Oli.Delivery.Sections.re_enroll_learner(user.id, section.id) do
+      {:ok, _} ->
+        {:noreply,
+         redirect(socket,
+           to:
+             Routes.live_path(
+               OliWeb.Endpoint,
+               OliWeb.Delivery.InstructorDashboard.InstructorDashboardLive,
+               socket.assigns.section.slug,
+               :manage
+             )
+         )}
+
+      _ ->
```

Comment:
Should we handle the error case? Perhaps with a flash message?

Reply:
Makes sense! Done :)